src/Mapping/Annotation/Tree.php

```diff
@@ -42,4 +45,21 @@ final class Tree extends Annotation
      * @deprecated to be removed in 4.0, unused, configure the property on the TreeRoot annotation instead
      */
     public $identifierMethod;
+
+    /**
+     * @phpstan-param class-string|null $type
+     */
+    public function __construct(array $data = [], ?string $type = null)
```

Comment:
All the public fields should be in the constructor, see

https://github.com/doctrine-extensions/DoctrineExtensions/blob/81458019f09ae999c4509e670f64594c6d8f7e3c/src/Mapping/Annotation/Timestampable.php#L26-L48